experimental/InjectiveResolutions/src/InjectiveResolutions.jl

```diff
@@ -1048,18 +1023,18 @@ false
     if is_zero(E)
       d = -1
     else 
-      d = dim(E)
+      d = krull_dim(E)
     end
-    cod = dim(R_B) - d
+    cod = krull_dim(R_B) - d
     if cod < j+n+3
         return false
     end               # S2 condition not satisfied
   end
 
   Jac = ideal(R, minors(map_entries(R, jacobian_matrix(gens(modulus(R)))), n))  # Compute minors of the Jacobian
-  d = dim(Jac)                   # Get dimension of the Jacobian
+  d = krull_dim(Jac)                   # Get dimension of the Jacobian
   d < 0 && (d = -Inf)            # Handle negative dimensions
```

Comment:
I am not going to change this as I did not test it, and this is existing code, not the "fault" of this PR -- but I think this might be causing type instability, as `-Inf` is a `Float64`; really should be using `-inf` here.
```suggestion
  d < 0 && (d = -inf)            # Handle negative dimensions
```

@emikelsons please in your PR, try making this change.

Reply:
Not sure how to best implement it, but mathematically it just wants to test if the module is zero. In that case, the formula on the next line only works if dim is `-inf`, that's why we are fixing this.  Some code in OSCAR also returns `-1` for the zero module.  Might be better to use one of the `is_zero` functions and then bypass the check of the formula?

Reply:
@tom111 are you referring to my comment? The value `d` is also used in the next line, so calling `krull_dim` like this seems perfectly reasonable. The only problem I see is the use of `-Inf` instead of `-inf`, which is of course easy to change -- but I might be missing something, so I don't want to apply it here untested in order to not hold up this PR further.

(Ah, seeing that the next thing the code does is to subtract it from some other `krull_dim`, perhaps even better would be to do
```suggestion
  d < 0 && return true        # Handle negative dimensions
```
but again I might be missing something -- we'll look into this in a follow-up PR :-) )